contributors/design-proposals/customresources-subresources.md

```diff
@@ -0,0 +1,221 @@
+}
+
+// CustomResourceSubResourceScale defines how to serve the HTTP path <CR name>/scale.
+type CustomResourceSubResourceScale struct {
+    // required, e.g. “.spec.replicas”.
+    // Only JSON paths without the array notation are allowed.
+    SpecReplicasPath string `json:“specReplicasPath,omitempty”`
+    // optional, e.g. “.status.replicas”.
+    // Only JSON paths without the array notation are allowed.
+    StatusReplicasPath string `json:“statusReplicasPath,omitempty”`
+    // optional, e.g. “.spec.labelSelector”.
+    // Only JSON paths without the array notation are allowed.
+    LabelSelectorPath string `json:“labelSelectorPath,omitempty”`
+}
+
+// The following is the payload to send over the wire for /scale. It happens
+// to be defined here in apiextensions.k8s.io because we don’t have a global
+// Scale type in meta/v1. Ref: https://github.com/kubernetes/kubernetes/issues/49504
+
+// Scale represents a scaling request for a resource.
+type Scale struct {
+    metav1.TypeMeta `json:",inline"`
+    // Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata.
+    // +optional
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // defines the behavior of the scale. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status.
+    // +optional
+    Spec ScaleSpec `json:"spec,omitempty"`
+
+    // current status of the scale. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status. Read-only.
+    // +optional
+    Status ScaleStatus `json:"status,omitempty"`
+}
+
+// ScaleSpec describes the attributes of a scale subresource.
+type ScaleSpec struct {
+    // desired number of instances for the scaled object.
+    // +optional
+    Replicas int32 `json:"replicas,omitempty"`
+}
+
+// ScaleStatus represents the current status of a scale subresource.
+type ScaleStatus struct {
+    // actual number of observed instances of the scaled object.
+    Replicas int32 `json:"replicas"`
+
+    // label query over pods that should match the replicas count.
+    // More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+    // +optional
+    Selector *metav1.LabelSelector `json:"selector"`
+}
+```
+
+### Feature Gate
+
+The `SubResources` field in `CustomResourceDefinitionSpec` will be gated under the `CustomResourceSubResources` alpha feature gate.
+If the gate is not open, the value of the new field within `CustomResourceDefinitionSpec` is rejected on creation and updates of CRDs.
+
+## Semantics
+
+### Validation Behavior
+
+#### Status
+
+The status endpoint of a CustomResource receives a full CR object. Changes outside of the status path are ignored.
+For validation everything outside of the JSON Path `StatusPath` can be reset to the existing values of the CustomResource. Then the JSON Schema presented in the CRD is validated against the whole object.
```

Comment:
It will still pass along the ResourceVersion from the update request to be verified, right? We need to be careful not to overwrite it with the old value.